File: src/main/java/io/github/oleksivio/telegram/bot/core/model/method/FormattedMessageAction.java
package io.github.oleksivio.telegram.bot.core.model.method;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.oleksivio.telegram.bot.core.controller.network.ActionNetworker;

public abstract class FormattedMessageAction extends MessageAction {
    /**
     * parse_mode String Optional Send Markdown or HTML, if you want Telegram apps to show bold,
     * italic, fixed-width text or inline URLs in your bot's message.
     */
    @JsonProperty("parse_mode")
    private String parseMode = null;

    protected FormattedMessageAction(String method, ActionNetworker actionNetworker) {
        super(method, actionNetworker);
    }

    protected void pSetParseMode(String parseMode) {
        this.parseMode = parseMode;
    }

    public String getParseMode() {
        return parseMode;
    }

    public abstract FormattedMessageAction setParseMode(String parseMode);

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/core/model/method/FormattedMessageAction.kt
package io.github.oleksivio.telegram.bot.core.model.method

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.telegram.bot.core.controller.network.ActionNetworker

abstract class FormattedMessageAction protected constructor(method: String, actionNetworker: ActionNetworker) : MessageAction(method, actionNetworker) {
    /**
     * parse_mode String Optional Send Markdown or HTML, if you want Telegram apps to show bold,
     * italic, fixed-width text or inline URLs in your bot's message.
     */
    @JsonProperty("parse_mode")
    var parseMode: String? = null

}
